### PR TITLE
Installing pg-16 instead of pg-14 in the installer script and also changed the pg_dump and pg_restore version being checked in code to 16

### PR DESCRIPTION
**Pull request by ShivanshGahlot**

### CODE REVIEW

File: installer_scripts/install-yb-voyager [hunk 1]
```diff
@@ -180,7 +180,7 @@ centos_main() {
 	output "Installing RPM dependencies."
 	$YUM which wget git gcc make 1>&2
 	$YUM https://download.postgresql.org/pub/repos/yum/reporpms/EL-${majorVersion}-x86_64/pgdg-redhat-repo-latest.noarch.rpm 1>&2 || true
-	$YUM postgresql14 1>&2
+	$YUM postgresql16 1>&2
```
Comment (makalaaneesh) [inline installer_scripts/install-yb-voyager:183]: Should we also be modifying voyager code to look for pg_dump>=16? cc @priyanshi-yb

File: installer_scripts/install-yb-voyager [hunk 2]
```diff
@@ -879,11 +879,10 @@ centos_install_oracle_instant_clients() {
 
 ubuntu_install_postgres() {
 	output "Installing postgres."
-	line="deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main"
-	sudo grep -qxF "$line" /etc/apt/sources.list.d/pgdg.list || echo "$line" | sudo tee /etc/apt/sources.list.d/pgdg.list 1>&2
-	wget -nv -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
+	sudo apt install -y postgresql-common 1>&2
```
Comment (makalaaneesh) [inline installer_scripts/install-yb-voyager:882]: what all does postgresql-common install?

Comment (makalaaneesh) [reply installer_scripts/install-yb-voyager:882]: Ah, is it just to be able to run the apt.postgresql.org.sh? 
and that script basically automates the steps like adding to pgdg.list, apt-key add, etc?

**Review by priyanshi-yb — APPROVED** [review]
LGTM

File: installer_scripts/install-yb-voyager [hunk 3]
```diff
@@ -879,11 +879,10 @@ centos_install_oracle_instant_clients() {
...
+    echo | sudo /usr/share/postgresql-common/pgdg/apt.postgresql.org.sh 1>&2
 	sudo apt-get update 1>&2
-	sudo apt-get -y install postgresql-14 1>&2
+	sudo apt-get -y install postgresql-16
```
Comment (makalaaneesh) [inline installer_scripts/install-yb-voyager:885]: @ShivanshGahlot we should add the `1>&2` back

**Pull request merged**